File: Console/STSimConsole.ListReports.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Imports SyncroSim.Core

Partial Class STSimConsole

    Public Sub New(ByVal session As Session, ByVal args As String())
        MyBase.New(session, args)
    End Sub

    Private Sub HandleListReportsArgument()

        If (Me.Help) Then
            System.Console.WriteLine("Lists available ST-Sim reports.")
        Else

            System.Console.WriteLine("Available reports:")
            System.Console.WriteLine()
            System.Console.WriteLine(STATECLASS_SUMMARY_REPORT_NAME)
            System.Console.WriteLine(TRANSITION_SUMMARY_REPORT_NAME)
            System.Console.WriteLine(TRANSITION_STATECLASS_SUMMARY_REPORT_NAME)
            System.Console.WriteLine(STATE_ATTRIBUTE_REPORT_NAME)
            System.Console.WriteLine(TRANSITION_ATTRIBUTE_REPORT_NAME)

        End If

    End Sub

End Class

File: src/Console/STSimConsole.ListReports.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using SyncroSim.Core;

namespace SyncroSim.STSim
{
    internal partial class STSimConsole
    {
        public STSimConsole(Session session, string[] args) : base(session, args)
        {
        }

        private void HandleListReportsArgument()
        {
            if (this.Help)
            {
                System.Console.WriteLine("Lists available ST-Sim reports.");
            }
            else
            {
                System.Console.WriteLine("Available reports:");
                System.Console.WriteLine();
                System.Console.WriteLine(Strings.STATECLASS_SUMMARY_REPORT_NAME);
                System.Console.WriteLine(Strings.TRANSITION_SUMMARY_REPORT_NAME);
                System.Console.WriteLine(Strings.TRANSITION_STATECLASS_SUMMARY_REPORT_NAME);
                System.Console.WriteLine(Strings.STATE_ATTRIBUTE_REPORT_NAME);
                System.Console.WriteLine(Strings.TRANSITION_ATTRIBUTE_REPORT_NAME);
            }
        }
    }
}
